src/utils.js

```diff
@@ -112,3 +60,84 @@ export function buildClause (field, value, opts) {
 
   return Object.assign({}, mainClause, opts)
 }
+
+export function toBool (filters) {
+  const unwrapped = {
+    must: unwrap(filters.and),
+    should: unwrap(filters.or),
+    must_not: unwrap(filters.not),
+    minimum_should_match: filters.minimum_should_match
+  }
+
+  if (
+    filters.and.length === 1 &&
+    !unwrapped.should &&
+    !unwrapped.must_not
+  ) {
+    return unwrapped.must
+  }
+
+  const cleaned = {}
+
+  if (unwrapped.must) {
+    cleaned.must = unwrapped.must
+  }
+  if (unwrapped.should) {
+    cleaned.should = filters.or
+  }
+  if (unwrapped.must_not) {
+    cleaned.must_not = filters.not
+  }
+  if (
+    unwrapped.minimum_should_match &&
+    unwrapped.minimum_should_match <= filters.or.length
+  ) {
+    cleaned.minimum_should_match = unwrapped.minimum_should_match
+  }
+
+  return {
+    bool: cleaned
+  }
+}
+
+function unwrap (arr) {
+  return arr.length > 1 ? arr : _.last(arr)
+}
+
+export function pushQuery (existing, boolType, filterType, ...args) {
```

Comment:
Optional: the naming of `filterType` can be confusing since this is used by query also. Rename to `queryType` or `clauseType` or just `type`.